### PR TITLE
add crossmodelmatrix

### PR DESCRIPTION
**Pull request by matthieugomez**
This is a  useful API for packages that compute robust standard errors. Different models may want to implement more efficient methods (accessors) since X'X is typically computed during the estimation.

### CODE REVIEW

File: docs/src/statmodels.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ fitted
 leverage
 meanresponse
 modelmatrix
+crossmodelmatrix
```
Comment (nalimilan) [inline docs/src/statmodels.md]: Please preserve lexicographic order.

File: src/statmodels.jl [hunk 2]
```diff
@@ -308,6 +308,14 @@ Return the model matrix (a.k.a. the design matrix).
 """
 modelmatrix(obj::RegressionModel) = error("modelmatrix is not defined for $(typeof(obj)).")
 
+
```
Comment (nalimilan) [inline src/statmodels.jl]:
```suggestion
```

File: src/statmodels.jl [hunk 3]
```diff
@@ -308,6 +308,14 @@ Return the model matrix (a.k.a. the design matrix).
...
+"""
+    crossmodelmatrix(obj::RegressionModel)
+
+Return X'X where X is the model matrix
```
Comment (nalimilan) [inline src/statmodels.jl]:
```suggestion
Return `X'X` where `X` is the model matrix of `obj`.
This function will return a pre-computed matrix stored in `obj` if possible.
```

**Comment by matthieugomez** [comment]
updated.

**Comment by matthieugomez** [comment]
Thanks!

**Pull request merged**